hexrdgui/masking/mask_manager.py

```diff
@@ -0,0 +1,390 @@
+                        final_mask = np.logical_and(final_mask, arr)
+                if self.view_mode != ViewType.raw:
+                    # Reset the masked arrays for the current view
+                    mask.update_masked_arrays(self.view_mode)
+            if tm := self.threshold_mask:
+                idx = HexrdConfig().current_imageseries_idx
+                thresh_mask = tm.masked_arrays[name][idx]
+                final_mask = np.logical_and(final_mask, thresh_mask)
+            masks_dict[name] = final_mask
+
+        return masks_dict
+
+    def setup_connections(self):
+        self.threshold_mask_changed.connect(self.threshold_toggled)
+        HexrdConfig().save_state.connect(self.save_state)
+        HexrdConfig().load_state.connect(self.load_state)
+        HexrdConfig().detectors_changed.connect(self.clear_all)
+        HexrdConfig().state_loaded.connect(self.rebuild_masks)
+
+    def view_mode_changed(self, mode):
+        self.view_mode = mode
+
+    def masks_changed(self):
+        if self.view_mode in (ViewType.polar, ViewType.stereo):
+            self.polar_masks_changed.emit()
+        elif self.view_mode == ViewType.raw:
+            self.raw_masks_changed.emit()
+
+    def rebuild_masks(self):
+        if self.view_mode == ViewType.raw:
+            rebuild_raw_masks()
+        elif self.view_mode in (ViewType.polar, ViewType.stereo):
+            rebuild_polar_masks()
+        self.masks_changed()
+        self.mask_mgr_dialog_update.emit()
+
+    def add_mask(self, name, data, mtype, visible=True):
+        # Enforce name uniqueness
+        name = unique_name(self.mask_names, name)
+        if mtype == MaskType.threshold:
+            new_mask = ThresholdMask(name, mtype, visible)
+        else:
+            new_mask = RegionMask(name, mtype, visible)
+        new_mask.set_data(data)
+        self.masks[name] = new_mask
+        self.mask_mgr_dialog_update.emit()
+        return new_mask
+
+    def remove_mask(self, name):
+        removed_mask = self.masks.pop(name)
+        self.mask_mgr_dialog_update.emit()
+        return removed_mask
+
+    def write_masks_to_group(self, data, h5py_group):
+        unwrap_dict_to_h5(h5py_group, data, asattr=False)
+
+    def write_single_mask(self, name):
+        d = {'_version': 2}
+        d[name] = self.masks[name].serialize()
+        self.export_masks_to_file.emit(d)
+
+    def write_all_masks(self, h5py_group=None):
+        d = {'_version': 2}
+        for name, mask_info in self.masks.items():
+            d[name] = mask_info.serialize()
+        if h5py_group:
+            self.write_masks_to_group(d, h5py_group)
+        else:
+            self.export_masks_to_file.emit(d)
+
+    def save_state(self, h5py_group):
+        if 'masks' not in h5py_group:
+            h5py_group.create_group('masks')
+
+        self.write_all_masks(h5py_group['masks'])
+
+    def load_masks(self, h5py_group):
+        items = load_masks_v1_to_v2(h5py_group)
```

Comment:
We can just call a `load_masks()` function here, and that function should figure out whether it should convert from `v1` to `v2` based upon the version number. (see previous comment about `load_masks()`.